packages/firebase_messaging/README.md

```diff
@@ -54,7 +54,74 @@ Note: When you are debugging on Android, use a device or AVD with Google Play se
       <category android:name="android.intent.category.DEFAULT" />
   </intent-filter>
   ```
+#### Optionally handle background messages
 
+By default background messaging is not enabled. To handle messages in the background:
+
+1. Add an Application.java class to your app
+
+```
+package io.flutter.plugins.firebasemessagingexample;
+
+import io.flutter.app.FlutterApplication;
+import io.flutter.plugin.common.PluginRegistry;
+import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback;
+import io.flutter.plugins.GeneratedPluginRegistrant;
+import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService;
+
+public class Application extends FlutterApplication implements PluginRegistrantCallback {
+  @Override
+  public void onCreate() {
+    super.onCreate();
+    FlutterFirebaseMessagingService.setPluginRegistrant(this);
+  }
+
+  @Override
+  public void registerWith(PluginRegistry registry) {
+    GeneratedPluginRegistrant.registerWith(registry);
+  }
+}
+```
+1. Set name property of application in `AndroidManifest.xml`
```

Comment:
This isn't showing up as an auto-numbered list (probably because of the intervening content) so you'll have to use 2, 3, etc.

Reply:
Had to indent the intervening content.